### DISCUSSION TITLE
WSACleanup() killing unit tests

### DISCUSSION THREAD

**Discussion by kartonka** [xunit/xunit#2539] [Question] [answered]
I've experienced difficulties while testing a dll function that calls WSACleanup(). The unit test system signals:

_"The active test run was aborted. Reason: Unable to communicate with test host process."_

After removing the line with _WSACleanup();_ the tests run successfully. I don't experience any problems if I execute the function from a console application.

I can't help but think that the unit test framework uses sockets and WSACleanup() affects them. Is it possible?

**Comment by bradwilson** [comment, marked as answer]
We don't use sockets, but if you're running us via VSTest (that is, via Visual Studio, Visual Studio Code, `dotnet test`, or `vstest.console.exe`), they may be using sockets for cross-process communication that you are aborting. I would suggest [asking your question of them](https://github.com/microsoft/vstest).
